crates/hir-ty/src/consteval.rs

```diff
@@ -169,15 +169,23 @@ pub fn usize_const(db: &dyn HirDatabase, value: Option<u128>, krate: CrateId) ->
 }
 
 pub fn try_const_usize(db: &dyn HirDatabase, c: &Const) -> Option<u128> {
+    try_const_usize_sign_extend(db, c, false)
+}
+
+pub fn try_const_usize_sign_extend(
+    db: &dyn HirDatabase,
+    c: &Const,
+    is_signed: bool,
+) -> Option<u128> {
```

Comment:
Let's just copy that function once for usize once for isize, the name here is otherwise rather confusing